src/functionalTest/java/uk/gov/hmcts/reform/draftstore/steps/DraftStoreSteps.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.draftstore.steps;
+
+import com.google.common.base.Strings;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import org.json.JSONException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.reform.draftstore.domain.DraftList;
+import uk.gov.hmcts.reform.draftstore.features.store.Draft;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+public class DraftStoreSteps {
+
+    @Value("${test-url}")
+    private String url;
+
+    @Autowired
+    private S2SAuthSteps s2SAuthSteps;
+
+    @Autowired
+    private IdamAuthSteps idamAuthSteps;
+
+    private Response response;
+
+    public List listDrafts() {
+        response = newRequest()
+            .get("/drafts")
+            .andReturn();
+
+        DraftList draftList = response.as(DraftList.class);
+        return draftList == null ? Collections.emptyList() : draftList.data;
+    }
+
+    private RequestSpecification newRequest() {
+        String s2s = s2SAuthSteps.authenticate().assertSuccessfulAuthentication().getJwt();
+        if (StringUtils.isEmpty(s2s)) {
+            throw new IllegalArgumentException("Empty S2S token");
+        }
+
+        String idam = idamAuthSteps.requestRegistration().authenticate().assertSuccessfulAuthentication().getJwt();
+        if (StringUtils.isEmpty(idam)) {
+            throw new IllegalArgumentException("Empty IdAM token");
+        }
+
+        return RestAssured.given()
+            .log().all()
+            .relaxedHTTPSValidation()
+            .baseUri(url)
+            .header("ServiceAuthorization", "Bearer " + s2s)
+            .header("Secret", Strings.repeat("x", 16))
+            .auth().oauth2(idam);
+    }
+
+    public Response lastResponse() {
```

Comment:
do you plan to make multiple http calls using you 'steps' object? atm this method is always used after single call. can the methods making a call return a response instead?